jwst/mrs_imatch/mrs_imatch_step.py

```diff
@@ -328,6 +329,31 @@ def _match_models(models, channel, degree, center=None, center_cs='image'):
     #if np.isnan(sigma_data).any():
     #    print('a nan exists in sigma data')
 
+    # MRS fields of view are small compared to source sizes,
+    # and undersampling produces significant differences
+    # in overlap regions between exposures.
+    # Therefore use sigma-clipping to detect and remove sources
+    # (and unmasked outliers) prior to doing the background matching.
+    # Loop over input exposures
+    for i in range(0,len(models)):
+        image = image_data[i]
+        mask = mask_data[i]
+        weight = sigma_data[i]
+        # Do statistics wavelength by wavelength
+        for j in range(0,len(image)):
+            thisimg = image[j,:,:]
+            thismask = mask[j,:,:]
+            thiswght = weight[j,:,:]
+            # Sigma clipped statistics, ignoring zeros where no data
+            _, themed, clipsig = sigclip(thisimg, mask_value=0.)
+            im1d = thisimg.ravel()
+            mask1d = thismask.ravel()
+            wght1d = thiswght.ravel()
+            # Reject beyond 3sigma
+            reject = np.where(np.abs(im1d - themed) > 3*clipsig)
+            im1d[reject] = themed
+            mask1d[reject] = 0
+            wght1d[reject] = 1e18
```

Comment:
I think masking should be enough (data and weights are ignored if mask is 0):
```suggestion
            mask1d[reject] = 0
```